.github/workflows/run-tests.yml

```
@@ -0,0 +1,74 @@
+      run: |
+        $testType = '${{ github.event.inputs.test_type }}'
+        if ([string]::IsNullOrEmpty($testType)) {
+          $testType = 'All'
+        }
+        
+        Write-Host "Running $testType tests..."
+        cd tests
+        ./Run-HomeLab-Tests.ps1 -TestType $testType -GenerateReport
+      
+    - name: Upload test results
+      uses: actions/upload-artifact@v3
```

actionlint (1.7.7)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)